pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -502,14 +502,15 @@ func (k *Kubernetes) CreateKubernetesObjects(name string, service kobject.Servic
 		log.Warning("Global mode not yet supported, containers will only be replicated once throughout the cluster. DaemonSet support will be added in the future.")
 		replica = 1
 	}
-
-	if opt.CreateD {
+	//fmt.Println("controller", opt.Controller, "provider", opt.Provider)
+	if opt.Controller == "deployment" && opt.Provider == "kubernetes" {
+		//fmt.Println("created")
 		objects = append(objects, k.InitD(name, service, replica))
 	}
-	if opt.CreateDS {
+	if opt.Controller == "daemonset" {
 		objects = append(objects, k.InitDS(name, service))
 	}
-	if opt.CreateRC {
+	if opt.Controller == "replicationcontroller" {
```

Comment:
you are checking against `replicationcontroller`. Whatever is coming from `opt.Controller` should be lowercased so that options such as `replicationController` work.